### ISSUE TITLE
`minute` and `second` temporal kernels do not respect timezone

### ISSUE THREAD

**Issue by alamb** [apache/arrow-rs#500]
**Describe the bug**
As pointed out by @jorgecarleitao  https://github.com/apache/arrow-rs/pull/493/files#r656987963 the `minute` and `second` temporal kernels do not respect timezone

**To Reproduce**
Extract seconds from a timestamp column that has an associated timezone (aka is not `None` or `Utc`) will always be interpreted as a timestamp in UTC rather than the specified timezone. 

**Expected behavior**
Seconds to be interpreted in terms of the the timezone or raise an error about not being supported

**Comment by sum12** [comment]
I would like to work on this.

split in two parts 
- if tz is of the form `[+-][0-9]{2}:[0-9]{2}` then construct a `chrono::offset::FixedOffset` object from the provided tz string and add that at offset to the `NaiveDateTime` (`NaiveDateTIme` implements `Add<FixedOffset>`)
- else if it is a string which cannot be parsed as above we use th `chrono-tz` crate to generate an offset 

sounds okay ?

**Comment by jorgecarleitao** [comment]
makes sense to me. I have done this recently in arrow2 around [here](https://github.com/jorgecarleitao/arrow2/blob/main/src/compute/cast/primitive_to.rs#L362)

**Comment by alamb** [comment]
@sum12  this also sounds reasonable to me. 

As you have probably seen the spec for what can be in the timezone string is here: https://github.com/apache/arrow/blob/master/format/Schema.fbs#L341-L351 and I believe your proposal in https://github.com/apache/arrow-rs/issues/500#issuecomment-915194224 is consistent with that

Regarding `chrono-tz`, please make that an optional dependency (via feature flag) of arrow-rs -- we have many users who use arrow in a variety of ways that are sensitive to additional dependencies.

**Comment by sum12** [comment]
Thanks for the feedback. 

@alamb,  created a PR for now, does not really follow the proposal.  I have tried to reuse the parsing capabilities of the chrono itself to parse the timezone. Not sure if it is okay to go down that route. Although happy to back to proposal if the PR is not going in the correct direction.
